### PR TITLE
Command Palette minor performance improvements

### PR DESCRIPTION
**Pull request by NickLanam**
Summary: While profiling a different improvement to the Command Palette, I found a few obvious spots to improve performance a little bit. These changes don't affect much as of this moment, but the change I extracted them from does need them.

Test Plan: Smoke test. Only change should be a marginally more responsive Command Palette while typing.

Type of change: /kind bug

Signed-off-by: Nick Lanam <nlanam@pixielabs.ai>

### CODE REVIEW

**Comment by JamesMBartlett** [comment]
This seems to include #1046 ?

**Comment by NickLanam** [comment]
Note: includes the changes from #1046 so that they can merge in either order instead of dealing with a rebase.

File: src/ui/src/components/command-palette/providers/script/script-provider-common.tsx [hunk 1]
```diff
@@ -36,22 +37,62 @@ export function quoteIfNeeded(input: string): string {
...
   highlights: number[],
 }>(({ icon, input, highlights }) => {
-  const outs: React.ReactNode[] = [];
-  for (let i = 0; i < input.length; i++) {
-    if (highlights.includes(i)) outs.push(<strong key={i}>{input.substring(i, i + 1)}</strong>);
-    else outs.push(<span key={i}>{input.substring(i, i + 1)}</span>);
-  }
+  const classes = useStyles();
+
+  const parts = React.useMemo(() => {
```
Comment (aimichelle) [inline src/ui/src/components/command-palette/providers/script/script-provider-common.tsx:68]: Can you just add a comment about what this is doing?

Comment (NickLanam) [reply src/ui/src/components/command-palette/providers/script/script-provider-common.tsx:69]: Done.

Comment (NickLanam) [reply src/ui/src/components/command-palette/providers/script/script-provider-common.tsx:69]: Done.

**Review by aimichelle — APPROVED** [review]

**Pull request merged**